core/src/math/BigInt.h

```diff
@@ -224,7 +224,15 @@ namespace ledger {
             BigInt& assignI64(int64_t value);
 
             template <typename T>
-            BigInt& assignScalar(T value) {
+            typename std::enable_if<std::is_unsigned<T>::value, BigInt&>::type assignScalar(T value) {
```

Comment:
I suggest you to read that [article](https://www.fluentcpp.com/2019/08/23/how-to-make-sfinae-pretty-and-robust/) about how prettify SFINAE.

For instance we can create generic type traits somewhere such as : 
```
template <typename T>
using isSigned = std::enable_if_t<std::is_signed<T>::value, bool>;

template <typename T>
using isUnsigned = std::enable_it_t<std::is_unsigned<T>::value, bool>;
```

then use it like that : 
```
template <typename T, isUnsigned<T> = true>
BigInt& assignScalar(T value)  {
...
}

template <typename T, isSigned<T> = true>
BigInt& assignScalar(T value) {
... 
}
```
The code is more readable and we can reuse these type traits for future implementations.

Reply:
Nice way! I updated the code.